internal/util/resource/resource.go

```diff
@@ -73,31 +76,26 @@ func Track(obj any, token *Token) {
 	}
 
 	// use token instead of obj itself,
-	// because otherwise profile will hold a reference to obj and finalizer will never run
+	// because otherwise profile will hold a reference to obj and cleanup will never run
 	p.Add(token, 1)
 
-	var stack string
+	token.msg = fmt.Sprintf("%T has not been finalized", obj)
```

Comment:
Why does `token.msg` need to be part of token struct :thinking:? It does not seem to be used outside this function.